consensus/state.go

```diff
@@ -411,6 +415,25 @@ func (cs *State) loadWalFile() error {
 
 // OnStop implements service.Service.
 func (cs *State) OnStop() {
+
+	// If the node is committing the new block, wait until it finished!
+	if cs.GetRoundState().Step == cstypes.RoundStepCommit {
+	wait:
+		for {
```

Comment:
Is a for loop necessary if you are breaking out of it in every case. AFAICS this only loops once

Reply:
You are right, I was thinking there are other events that will fall into the first case but didn't meet the condition.